### PR TITLE
feat(config): upgrade to angular 1.5

### PR DESCRIPTION
**Pull request by trendzetter**
Upgrade dependencies to more recent versions.
Migration guide: https://docs.angularjs.org/guide/migration

### CODE REVIEW

**Comment by trendzetter** [comment]
I think I tested all functionality except authorization using external providers. All seems good.

File: bower.json [hunk 1]
```diff
@@ -4,19 +4,19 @@
...
+    "angular-mocks": "~1.4",
+    "angular-resource": "~1.4",
     "angular-ui-router": "~0.2",
     "angular-ui-utils": "bower",
     "bootstrap": "~3",
     "owasp-password-strength-test": "~1.3.0"
   },
   "resolutions": {
-    "angular": "~1.3"
+    "angular": "~1.4"
```
Comment (ilanbiala) [inline bower.json]: Why is this necessary? Can we fix the dependencies so the resolution is unnecessary.

**Comment by ilanbiala** [comment]
@trendzetter our client-side tests probably don't catch everything. Have you tried running the app as well to make sure nothing has broken on the frontend?

**Comment by trendzetter** [comment]
@ilanbiala I did run the app and tested the functionality except external authorisation. All seems unchanged. As concerning the resolution, I just replaced an existing hard-coded resolution. Fixing the resolution should be a different PR I think because it involves changing unrelated things such as adding kerberos to packages.json.
see also here: https://github.com/meanjs/mean/pull/1174

**Comment by ilanbiala** [comment]
Angular doesn't rely on Kerberos. One of the bower dependencies requires a different version, and we should try to update that so it works with Angular 1.4.

**Comment by trendzetter** [comment]
The resolution was broken with 1.3 and I can't fix it without fixing the whole issue with dependencies. Specifically mongo depends on kerberos. The existing hard-coded resolution forces bower to continue ignoring dependency errors. It's just an existing issue that needs attention but it's not specific to this pr.

**Comment by trendzetter** [comment]
I looked further into it and kerberos has nothing to do with this. I am mistaken

**Comment by trendzetter** [comment]
Resolved the dependency by updating file upload. Functionality is working.

**Comment by rhutchison** [comment]
Why not 1.5?

**Comment by trendzetter** [comment]
1.4 is set in the goals for the milestone and it's a fully matured version but I too suspect that 1.5.0, a bit fresh, would work just as well. Update the PR?

**Comment by rhutchison** [comment]
@trendzetter up to the community, but I would imagine when the milestone was written, 1.5.0 wasn't released.

@codydaig @lirantal @ilanbiala @mleanos @trainerbill @vaucouleur

**Comment by ilanbiala** [comment]
1.5.0 is fine with me as long as there are still no functionality issues and everything works for us as expected. @codydaig any objections?

**Comment by trendzetter** [comment]
functionality works as expected after upgrading to 1.5 too.

**Comment by mleanos** [comment]
@trendzetter The client-side tests are failing. It could have something to do with the package upgrades. 

Can you check on that locally, and make sure all the tests (both client & server) are running?

**Comment by trendzetter** [comment]
I have no clue why the tests are failing.

**Comment by trendzetter** [comment]
I could revert to 1.4?

**Comment by rhutchison** [comment]
No need to revert - we can help out

**Comment by ilanbiala** [comment]
@trendzetter can you log what `ArticlesService` is and look into why it doesn't fail on https://github.com/meanjs/mean/blob/master/modules/articles/tests/client/articles.client.controller.tests.js#L50

**Comment by rhutchison** [comment]
@trendzetter I pulled your branch locally and the tests run fine...  Does it fail in your local env?

**Comment by lirantal** [comment]
@rhutchison you can look at the travis log to see why it's failing too: https://travis-ci.org/meanjs/mean/builds/108896432

**Comment by mleanos** [comment]
I pulled this branch down as well, and the Karma tests are failing for me, with unreadable Angular errors. My best guess is that it's one of the package upgrades here. I'll do some more debugging, but the errors aren't typical of something inside the framework.

Node: `v0.12.7`
NPM: `2.11.3`

**Comment by trendzetter** [comment]
Testing seems changed in 1.5:
http://puigcerber.com/2016/02/07/how-to-test-angular-1-5-components/

Can it be that we need to inject the article service rather than instantiate it?
https://github.com/Puigcerber/angular-unit-testing#using-provide

File: config/assets/default.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ module.exports = {
         'public/lib/angular-ui-router/release/angular-ui-router.js',
         'public/lib/angular-ui-utils/ui-utils.js',
         'public/lib/angular-bootstrap/ui-bootstrap-tpls.js',
```
Comment (zargold) [inline config/assets/default.js]: Having applied this, the problem is here: you also need and it has to be first `'public/lib/angular-bootstrap/ui-bootstrap.js'`

File: config/assets/production.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ module.exports = {
         'public/lib/angular-ui-router/release/angular-ui-router.min.js',
         'public/lib/angular-ui-utils/ui-utils.min.js',
```
Comment (zargold) [inline config/assets/production.js]: `'public/lib/angular-bootstrap/ui-bootstrap.js'`

**Comment by ilanbiala** [comment]
@trendzetter @usta can you guys work together on one PR (either this one #1207 or #1216) and try to solve the client-side dependency issue with Angular? Please figure out which one to use and close the other one as it's just a duplicate.

**Comment by trendzetter** [comment]
Since today all the changes I proposed are incorporated in the pr of ursa. I don't care too much to have my name on it so I will close this pr.

**Comment by usta** [comment]
it was my mistake to not to check  if any similar PR exist or not. So it will be more ethical if the pr will be @trendzetter 's.
Whenever we come to discuss test fails exist right now  there is 2 problem:
1- on serverside we need to upgrade to phantomjs-prebuilt "phantomjs-prebuilt": "~2.1.4", and  "karma-phantomjs-launcher": "~1.0.0",
2- after first step done we need to rewrite chat module's tests or we need to remove use strict from this two : socket.io.client.service.tests.js and chat.client.controller.tests.js
Then everything looks good

**Comment by ilanbiala** [comment]
@usta can you work with @trendzetter to take care of those two things on one of the PRs? It doesn't matter which one, if you'd like to do it on this one that's fine.

**Comment by usta** [comment]
@trendzetter could you reopen this pr or create a new one. Please because it is more ethical for me

File: config/assets/default.js (hunk 2 above)
Comment (usta) [inline config/assets/default.js]: @zargold  i don't think that one is correct . ( I mean current pr look good ) Because : ""Now it should be clear that files with the -tpls in their name have bootstrap-specific templates bundled with directives. For people who want to take all the directives and don't need to customize anything the solution is to grab a file named ui-bootstrap-tpls-[version].min.js. If, on the other hand default templates are not what you need you could take ui-bootstrap-[version].min.js and provide your own templates""

Comment (zargold) [inline config/assets/default.js]: If you don't have this correction then you must remove dependency injection `['ui.bootstrap']` and the dropdowns won't work... but you know no big deal otherwise you will get error [ui.bootstrap] is not defined.

**Comment by trendzetter** [comment]
I could try to take a look but it can take a few days or even longer before I can really get into the unit testing in angular. No need to wait for me if you know how to get it going now.

File: config/assets/default.js (hunk 2 above)
Comment (trendzetter) [inline config/assets/default.js]: Have you actually tried because it seems to work just fine for me. No errors, dropdown working.

Comment (usta) [inline config/assets/default.js]: yeah it was working perfect without public/lib/angular-bootstrap/ui-bootstrap.js' also it is written in its own documentations that we need to decide  (ui-bootstrap.js  and some of templates ) or ui-bootstrap-tpls.js

**Comment by trendzetter** [comment]
I have rebase on the current master so the server dependencies should be ok now. I am still getting those unit test errors. How should we continue?

**Comment by usta** [comment]
As i said that before : 
1- on serverside we need to upgrade to phantomjs to "phantomjs-prebuilt": "~2.1.4", and  update karma-phantomjs-launcher to "karma-phantomjs-launcher": "~1.0.0",
2- after first step done we need to rewrite chat module's tests or we need to remove use strict from this two : socket.io.client.service.tests.js and chat.client.controller.tests.js
Then everything looks good But Second Step(No2) can be skipped now.
BUT This 2 are just serverside test falsepossitives so it is not this PRs problems

**Comment by trendzetter** [comment]
@usta I do read your comments but I only understand them vaguely. 
Are you suggesting to create a new pr which updates phantomjs-prebuilt and karma-phantomjs-launcher in packages.json? Or better include it here?
I tried those changes just now and I am indeed getting new errors on the chat module.

**Comment by ilanbiala** [comment]
Anything that relates to upgrading Angular should be in this PR. If you upgrade other dependencies, those should be in this same PR. Fixing the tests should also be in this PR>

**Comment by usta** [comment]
@trendzetter  Hello , first step is to add that step 1 ( phantomjs to "phantomjs-prebuilt": "~2.1.4", and update karma-phantomjs-launcher to "karma-phantomjs-launcher": "~1.0.0",  ) in this PR. And second step we need to rewrite meanjs/mean/modules/chat/tests/client/chat.client.controller.tests.js
and mean/modules/core/tests/client/socket.io.client.service.tests.js  tests .
On the other hand i'm really a newbie about writing tests.Maybe @ilanbiala  can help us about rewriting these test.
I believe these will finish all requests to release 0.5.0

**Comment by usta** [comment]
@trendzetter by the way sorry about my anger in previous comments. I was in a really bad mood while writing those comments. I hope you didn't get sad or feed bad about it. Sorry again

**Comment by trendzetter** [comment]
I have updated packages.json. The resulting errors should be visible on travis shortly for everyone to suggest a way to fix the tests.

**Comment by ilanbiala** [comment]
@trendzetter try using this code for the socket.io test file:

```
(function(window) {
  'use strict';

  /* Creates a mock of socket.io for the browser.
   * Functionality of the service is tested through
   * the chat controller tests.
   */
  var mock = function () {
    var io = {
      cbs: {},
      connect: connect,
      emit: emit,
      on: on,
      removeListener: removeListener
    };

    connect();

    return io;

    function connect() {
      io.socket = {};
    }

    function emit(msg, data) {
      io.cbs[msg](data);
    }

    function on(msg, cb) {
      io.cbs[msg] = cb;
    }

    function removeListener(msg) {
      delete io.cbs[msg];
    }
  };
  
  window.io = mock;
})(window);
```

**Comment by usta** [comment]
@ilanbiala this fixes all test fails but still got this one #1196 Because i'm getting that error in my mongodb in windows systems

**Comment by usta** [comment]
@trendzetter please change mean/modules/core/tests/client/socket.io.client.service.tests.js as @ilanbiala said

**Comment by trendzetter** [comment]
works for me.

**Comment by usta** [comment]
by the way i'm using :
D:\mongodb\bin>mongo.exe --version
MongoDB shell version: 3.3.1

D:\mongodb\bin>mongod.exe --version
db version v3.3.1
git version: e1a38c10c8e38f15af891ed330a31636f9e3857f
OpenSSL version: OpenSSL 1.0.1p-fips 9 Jul 2015
allocator: tcmalloc
modules: none
build environment:
    distmod: 2008plus-ssl
    distarch: x86_64
    target_arch: x86_64

and getting :
  1) User CRUD tests should not be able to update own user details with existing username:

      Uncaught AssertionError: expected '11000 duplicate key error collection: mean-test.users index: username already exists' to be 'Username already exists'
      + expected - actual

      -11000 duplicate key error collection: mean-test.users index: username already exists
      +Username already exists



  2) User CRUD tests should not be able to update own user details with existing email:

      Uncaught AssertionError: expected '11000 duplicate key error collection: mean-test.users index: email already exists' to be 'Email already exists'
      + expected - actual

      -11000 duplicate key error collection: mean-test.users index: email already exists
      +Email already exists

**Comment by usta** [comment]
Also adding this fixes all things:
diff --git a/modules/users/tests/server/user.server.routes.tests.js b/modules/users/tests/server/user.server.routes.tests.js
index 5d57c65..e088e9f 100644
--- a/modules/users/tests/server/user.server.routes.tests.js
+++ b/modules/users/tests/server/user.server.routes.tests.js
@@ -742,7 +742,7 @@ describe('User CRUD tests', function () {
               }

               // Call the assertion callback
-              userInfoRes.body.message.should.equal('Username already exists');
+              userInfoRes.body.message.toLowerCase().should.containEql('username already exists');

               return done();
             });
@@ -794,7 +794,7 @@ describe('User CRUD tests', function () {
               }

               // Call the assertion callback
-              userInfoRes.body.message.should.equal('Email already exists');
+              userInfoRes.body.message.toLowerCase().should.containEql('email already exists');

               return done();
             });

**Comment by usta** [comment]
I think this will also works with older versions. It will be much better to fix thisbefore someother people also hit this bug

**Comment by usta** [comment]
@trendzetter could you add the latest patch which i mentioned at https://github.com/meanjs/mean/pull/1207#issuecomment-189020292 Then i think all of these solve whole problems in both server and client side about angular 1.5

**Comment by mleanos** [comment]
@trendzetter @usta Thank you both for the hard work that was put into getting this PR worked out. I pulled this branch down, and tested. Everything seems to be working as expected. All the tests are passing for me, and I didn't notice any issues when I did a quick smoke check; with the exception of one UI glitch/issue.

When doing a hard refresh (CTRL+F5), while **not** logged in, I get a quick flash of a horizontal scroll-bar in the header & it disappears after about 1.5 secs. I get similar behavior when logged in, but no scroll-bar pops up; just slow refresh of the header menu after the rest of the page has already finished loading.

gh-1207-scroll-bar-in-header

Any idea what could be causing this?

**Comment by trendzetter** [comment]
@mleanos I also noticed this effect but it's gone so fast that I didn't pay much attention to it. 
@usta will try your change tonight.

**Comment by usta** [comment]
@ilanbiala could you please wait until @trendzetter applies latest patch which i wanted before merging this PR.

**Comment by ilanbiala** [comment]
I wasn't planning on merging just yet, I was just adding some metadata about the PR.

**Comment by trendzetter** [comment]
We should look into the remark of @mleanos . It's a minor issue but still a stain.

File: modules/users/tests/server/user.server.routes.tests.js [hunk 4]
```diff
@@ -742,7 +742,7 @@ describe('User CRUD tests', function () {
               }
 
               // Call the assertion callback
-              userInfoRes.body.message.should.equal('Username already exists');
+              userInfoRes.body.message.toLowerCase().should.containEql('username already exists');
```
Comment (ilanbiala) [inline modules/users/tests/server/user.server.routes.tests.js]: Please add a comment for this line and L797 explaining what string MongoDB 2.x and what string MongoDB 3.x expect so it's clear why this change was made.

**Comment by ilanbiala** [comment]
@trendzetter @usta do these tests not fail for you with MongoDB 3.x?

```
1) Users E2E Tests: Signup Validation Should report Email already exists
   Message:
     Expected '11000 duplicate key error collection: mean-test.users index: email already exists' to be 'Email already exists'.
   Stacktrace:
     Error: Failed expectation
    at [object Object].<anonymous> (mean/modules/users/tests/e2e/users.e2e.tests.js:275:62)
    at process._tickCallback (node.js:402:9)

  2) Users E2E Tests: Signup Validation Should report Username already exists
   Message:
     Expected '11000 duplicate key error collection: mean-test.users index: username already exists' to be 'Username already exists'.
   Stacktrace:
     Error: Failed expectation
    at [object Object].<anonymous> (mean/modules/users/tests/e2e/users.e2e.tests.js:294:62)
    at process._tickCallback (node.js:402:9)
```

**Comment by trendzetter** [comment]
@ilanbiala  I am using mongodb 2.6, @usta has mongodb 3 it appears.

**Comment by ilanbiala** [comment]
Yes, I believe these 2 tests fail on MongoDB 3. @usta can you see about the E2E tests when run with MongoDB 3?

**Comment by mleanos** [comment]
I'm using MongoDB version `3.0.7` and don't experience this issue of those E2E tests failing. I don't think they changed the messages until a later version.

@trendzetter Even though the glitch I reported isn't a major (functional) issue, it's still an undesired UI problem. I won't have much time today, but perhaps tomorrow I'll take a look at it. It's more than likely an issue with the Menu/Header.

**Comment by simison** [comment]
You guys clearly need this: https://www.npmjs.com/package/mongoose-beautiful-unique-validation

**Comment by trendzetter** [comment]
Just poking around to see if I can get some clues. Not that I propose this as a solution. If I leave .navbar-right out it is no longer showing showing the scrollbar for a moment. The menu text is still blinking on load if you look closely.

**Comment by trendzetter** [comment]
@mleanos My suspicion now is that we have a conflict between angular-bootstrap and bootstrap:
https://github.com/angular-ui/bootstrap/issues/2156

**Comment by ilanbiala** [comment]
@trendzetter that issue is already closed, what is the conflict?

**Comment by trendzetter** [comment]
@ilanbiala It's not the a conflict with the dropdown it appears. For me leaving out the redundant collapse class solves the issue.

**Comment by trendzetter** [comment]
@mleanos Can you confirm that it is solved?

**Comment by mleanos** [comment]
@trendzetter That solves the issue for me as well. Thank you!

This LGTM now.

File: modules/users/tests/server/user.server.routes.tests.js (hunk 4 above)
Comment (ilanbiala) [inline modules/users/tests/server/user.server.routes.tests.js]: @trendzetter just add a comment for ^^ and I'll merge it in.

**Comment by ilanbiala** [comment]
@trendzetter @usta any issue with these tests (https://github.com/meanjs/mean/pull/1207#issuecomment-189416818) on MongoDB 3?

**Comment by usta** [comment]
@ilanbiala i haven't time to test latest shape of PR but it is exactly same in my system and it doesn't have any fails or mongodb test errors. So it is OK :)

File: modules/users/tests/server/user.server.routes.tests.js (hunk 4 above)
Comment (mleanos) [inline modules/users/tests/server/user.server.routes.tests.js]: @trendzetter FYI, I'm using MongoDB `3.0.7` & don't have this issue. So perhaps indicate what exact version, or clarify that some release after `3.0.7` changes the validation messages, in your comment here.

Comment (usta) [inline modules/users/tests/server/user.server.routes.tests.js]: @mleanos : I have mentioned the versions but maybe you missed that comment:
D:\mongodb\bin>mongo.exe --version
MongoDB shell version: 3.3.1

D:\mongodb\bin>mongod.exe --version
db version v3.3.1
git version: e1a38c10c8e38f15af891ed330a31636f9e3857f
OpenSSL version: OpenSSL 1.0.1p-fips 9 Jul 2015
allocator: tcmalloc
modules: none
build environment:
distmod: 2008plus-ssl
distarch: x86_64
target_arch: x86_64

**Comment by trendzetter** [comment]
@ilanbiala I got mongodb 3.2.3 installed now. No issues with the tests found.

**Comment by usta** [comment]
@trendzetter could you please add a single line comment under mongodb test lines we add those workaround hack because of mongodb >3.2 have changed their validation output.

**Comment by usta** [comment]
:+1:

**Comment by trendzetter** [comment]
Any other remarks?

**Comment by ilanbiala** [comment]
LGTM, @codydaig @lirantal anything else before I merge?

File: modules/core/tests/client/socket.io.client.service.tests.js [hunk 5]
```diff
@@ -1,24 +1,39 @@
-(function() {
+(function(window) {
```
Comment (codydaig) [inline modules/core/tests/client/socket.io.client.service.tests.js]: @ilanbiala When are we passing in window and when do we not? It seems to be mixed in recent PRs.

**Comment by codydaig** [comment]
@ilanbiala Left one line comment. Otherwise, LGTM.

File: modules/core/tests/client/socket.io.client.service.tests.js (hunk 5 above)
Comment (ilanbiala) [inline modules/core/tests/client/socket.io.client.service.tests.js]: I'm not really sure when, maybe the style guide has some info?

Comment (trendzetter) [inline modules/core/tests/client/socket.io.client.service.tests.js]: Can point me to docs setting out those rules? Why would you pass window when it is not required for the workings of the code?

Comment (ilanbiala) [inline modules/core/tests/client/socket.io.client.service.tests.js]: https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md will have all the rules we pretty much follow.

File: bower.json [hunk 6]
```diff
@@ -4,19 +4,16 @@
...
-    "bootstrap": "~3",
+    "angular": "~1.5.0",
+    "angular-animate": "~1.5.0",
+    "angular-bootstrap": "~1.1.2",
+    "angular-file-upload": "~2.2.0",
+    "angular-messages": "~1.5.0",
+    "angular-mocks": "~1.5.0",
+    "angular-resource": "~1.5.0",
+    "angular-ui-router": "~0.2.18",
+    "angular-ui-utils": "~0.1.1",
```
Comment (simison) [inline bower.json]: BTW `angular-ui-utils` is deprecated: https://github.com/angular-ui/ui-utils

Comment (trendzetter) [inline bower.json]: removed angular-ui-utils

**Comment by trendzetter** [comment]
@simison removed angular-ui-utils

@ilanbiala @codydaig Are you ok with the removal of the deprecated dependency we are not using?

File: modules/core/tests/client/socket.io.client.service.tests.js (hunk 5 above)
Comment (trendzetter) [inline modules/core/tests/client/socket.io.client.service.tests.js]: Thanks @ilanbiala , it's pretty elaborate :) So something needs to change?

**Comment by trendzetter** [comment]
@codydaig I modified the updated test to inject window rather than passing it in and out. Is this what you are suggesting, if not could you be more specific?

**Comment by mleanos** [comment]
LGTM. Let's see what Cody has to say about injecting `window`. I'm not sure of the reasons to do it one way, or the other.

**Comment by trendzetter** [comment]
It could be an improvement to use the $window service rather than the window object. I don't know what John Papa says but I would be curious to learn, still digging through the guide.

**Comment by trendzetter** [comment]
https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y180

**Comment by ilanbiala** [comment]
Yeah $window should be used. Also, I'm fine with removing unused dependencies.

**Comment by codydaig** [comment]
Using $window works great. I'm also fine with removing unused dependencies.

**Comment by usta** [comment]
@trendzetter can you try to upgrade angular-bootstrap to 1.2.1 ? is there any problem with it or not ?  And if there isn't any more problems after upgrading to1.2.1 lets we merge this because this PR waited more than it needs to be.

**Comment by trendzetter** [comment]
it is only released since 4 days that is why it is not yet included. I will check it tonight.

File: bower.json [hunk 7]
```diff
@@ -4,19 +4,15 @@
...
-    "angular-file-upload": "1.1.5",
-    "angular-messages": "~1.3.17",
-    "angular-mocks": "~1.3",
-    "angular-resource": "~1.3",
-    "angular-ui-router": "~0.2",
-    "angular-ui-utils": "bower",
-    "bootstrap": "~3",
+    "angular": "~1.5.0",
+    "angular-animate": "~1.5.0",
+    "angular-bootstrap": "~1.2.1",
```
Comment (trendzetter) [inline bower.json]: Updated to the latest greatest!

**Comment by usta** [comment]
@ilanbiala can we merge this now ? It looks everything perfect. @mleanos @lirantal @codydaig

**Comment by mleanos** [comment]
LGTM. I haven't tested the latest changes, but if the tests pass then this is probably ready to be merged.

@trendzetter You did a quick smoke check, after upgrading `angular-bootstrap`, to make sure there are no adverse affect on the UI?

Thank you very much for working together on this, @trendzetter & @usta. It really helped streamline this upgrade.

**Comment by ilanbiala** [comment]
@trendzetter just confirm that you manually tested this as well, and I'll merge in.

**Comment by trendzetter** [comment]
@ilanbiala I ran this manually and all looks good.

**Comment by QiyuLi** [comment]
I think there is a small issue with MongoDB 3.2.
You PR fixed the incompatibility in CRUD test but not e2e test.

Here is the similar error:

Failures:

  1) Users E2E Tests: Signup Validation Should report Email already exists
   Message:
     Expected '11000 duplicate key error collection: mean-test.users index: email already exists' to be 'Email already exists'.
   Stacktrace:
     Error: Failed expectation
    at [object Object].<anonymous> (/vagrant/meanjs/modules/users/tests/e2e/users.e2e.tests.js:275:62)
    at process._tickCallback (node.js:364:9)

  2) Users E2E Tests: Signup Validation Should report Username already exists
   Message:
     Expected '11000 duplicate key error collection: mean-test.users index: username already exists' to be 'Username already exists'.
   Stacktrace:
     Error: Failed expectation
    at [object Object].<anonymous> (/vagrant/meanjs/modules/users/tests/e2e/users.e2e.tests.js:294:62)
    at process._tickCallback (node.js:364:9)

Finished in 77.414 seconds
27 tests, 30 assertions, 2 failures

**Comment by trendzetter** [comment]
Strange, I didn't get those errors. I have mongodb3.2.3 locally.

**Comment by ilanbiala** [comment]
@trendzetter I'm reverting the merge, can you fix those errors the same way you fixed them for the server and let me know when it's fixed?

**Comment by trendzetter** [comment]
Finished in 70.057 seconds
27 tests, 30 assertions, 0 failures

**Comment by trendzetter** [comment]
Is someone able to reproduce this error?

**Comment by ilanbiala** [comment]
@trendzetter I've had it a couple times, can you just fix those assertions to be looser in a new PR?

**Comment by QiyuLi** [comment]
@trendzetter @ilanbiala 

I'm reading Mongoose doc. I did not find any info about configuring err msg for unique.
I'm running Ubuntu 14.04 with MongoDB 3.2.3.

I attached a screenshot. This is not good to display such a detailed error.
image
image
